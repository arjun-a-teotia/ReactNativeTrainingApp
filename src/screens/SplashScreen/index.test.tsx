/**
 * @format
 */

import 'react-native';
import React from 'react';
import {SplashScreen} from './index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {waitFor} from '@testing-library/react-native';
import analytics from '@react-native-firebase/analytics';
const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actual = jest.requireActual('@react-navigation/native');

  return {
    ...actual,
    useNavigation: () => ({
      navigate: mockNavigate,
      dispatch: mockNavigate,
    }),
  };
});

jest.mock('@react-native-firebase/app', () => {
  return () => ({
    app: jest.fn(),
  });
});
jest.mock('@react-native-firebase/auth', () => {
  return jest.fn().mockReturnValue({currentUser: jest.fn()});
});
jest.mock('@react-native-firebase/analytics', () => {
  return jest.fn().mockReturnValue({logEvent: jest.fn()});
});
jest.mock('react-native-qrcode-svg', () => ({}));
jest.useFakeTimers();
describe('Splash Screen Testing', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SplashScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('wait for analytics calling', async () => {
    await waitFor(async () => {
      expect(analytics().logEvent).toBeCalledTimes(1);
    });
  });
  it('navigates to profile screen after 1 sec', async () => {
    await waitFor(() => {
      expect(mockNavigate).toHaveBeenCalledTimes(1);
    });
  });
});
