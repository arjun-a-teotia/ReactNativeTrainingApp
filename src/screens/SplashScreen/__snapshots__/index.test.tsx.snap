// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen Testing renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Welcome to IW Training App
  </Text>
  <ActivityIndicator
    size="large"
  />
</RNCSafeAreaView>
`;
